hoot-services/src/main/java/hoot/services/command/ExternalCommandRunnerImpl.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.commons.exec.PumpStreamHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import hoot.services.models.db.CommandStatus;
+import hoot.services.models.db.QCommandStatus;
+
+
+/**
+ * Utility class for running a subprocess synchronously from Java.
+ */
+public class ExternalCommandRunnerImpl implements ExternalCommandRunner {
+    private static final Logger logger = LoggerFactory.getLogger(ExternalCommandRunnerImpl.class);
+
+    private ExecuteWatchdog watchDog;
+    private OutputStream stdout;
+    private OutputStream stderr;
+
+    public ExternalCommandRunnerImpl() {}
+
+    @Override
+    public CommandResult exec(String[] command, String jobId, String caller) {
+        logger.debug("About to execute the following command: {}", commandArrayToString(command, caller));
+
+        try (OutputStream stdout = new ByteArrayOutputStream();
+             OutputStream stderr = new ByteArrayOutputStream()) {
+
+            this.stdout = stdout;
+            this.stderr = stderr;
+
+            ExecuteStreamHandler executeStreamHandler = new PumpStreamHandler(stdout, stderr);
+            Executor executor = new DefaultExecutor();
+            this.watchDog = new ExecuteWatchdog(ExecuteWatchdog.INFINITE_TIMEOUT);
+            executor.setWatchdog(this.watchDog);
+            executor.setStreamHandler(executeStreamHandler);
+
+            LocalDateTime start = null;
+            Exception exception = null;
+            int exitCode;
+
+            try {
+                CommandLine cmdLine = new CommandLine(command[0]);
+                for (int i = 1; i < command.length; i++) {
+                    cmdLine.addArgument(replaceSensitiveData(command[i]), false);
+                }
+
+                start = LocalDateTime.now();
+
+                logger.debug("Command {} started at: {}", commandArrayToString(command, caller), start);
+
+                exitCode = executor.execute(cmdLine);
+            }
+            catch (Exception e) {
+                exitCode = CommandResult.FAILURE;
+                exception = e;
+            }
+
+            if (executor.isFailure(exitCode) && this.watchDog.killedProcess()) {
+                // it was killed on purpose by the watchdog
+                logger.info("Process for {} command was killed!", commandArrayToString(command, caller));
+            }
+
+            LocalDateTime finish = LocalDateTime.now();
+
+            //, exitCode, stdout.toString(), stderr.toString()
+            CommandResult commandResult = new CommandResult();
+            commandResult.setCommand(commandArrayToString(command, caller));
+            commandResult.setExitCode(exitCode);
+            commandResult.setStderr(stderr.toString());
+            commandResult.setStdout(stdout.toString());
+            commandResult.setStart(start);
+            commandResult.setFinish(finish);
+            commandResult.setJobId(jobId);
+
+            updateDatabase(commandResult);
```

Comment:
Right now, we don't have any logic to truncate/prune the table when it grows too large in size.  We can always add a custom service within Services to always keep the table's size in check by leaving  only the last n command_status records.  This can also be a DB maintenance step by someone in charge of a Hootenanny instance.